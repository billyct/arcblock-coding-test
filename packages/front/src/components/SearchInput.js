import {forwardRef} from 'react'

const SearchInput = (props, ref) => (
  <input
    className='pl-4 pr-10 outline-none font-light h-full w-full'
    ref={ref}
    {...props}
  />
)

export default forwardRef(SearchInput)
