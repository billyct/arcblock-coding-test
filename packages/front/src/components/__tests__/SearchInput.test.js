import {createRef} from 'react'
import {render} from '@testing-library/react'

import SearchInput from '../SearchInput'

describe('test SearchInput Component', () => {
  it('should render correct', function () {
    const {container} = render(<SearchInput/>)
    expect(container).toMatchSnapshot()
  })

  it('should ref with the input', function () {
    let ref = createRef()
    const {getByRole} = render(<SearchInput ref={ref}/>)
    expect(ref.current).toEqual(getByRole('textbox'))
  })
})

