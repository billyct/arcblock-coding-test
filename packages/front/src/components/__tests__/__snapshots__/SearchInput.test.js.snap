// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchInput Component should render correct 1`] = `
<div>
  <input
    class="pl-4 pr-10 outline-none font-light h-full w-full"
  />
</div>
`;
